base/deprecated.jl

```diff
@@ -1763,4 +1763,10 @@ end)
     return false
 end
 
+# Not exported
+function LibGit2.owner(x::Union{LibGit2.GitIndex, LibGit2.GitTree, LibGit2.GitReference})
+    depwarn("owner(x) should be replaced with repository(x)", :owner)
+    LibGit2.repository(x)
+end
```

Comment:
For consistency with other deprecations, I would do
```julia
function LibGit2.owner(x)
    depwarn("owner(x) is deprecated, use repository(x) instead.", :owner)
    LibGit2.repository(x)
end
```
Note that I've left out the specification of the input type. I think it's unnecessary in this case, because existing code shouldn't be affected and new code will get a `MethodError` from `repository` for any weird input.

I'm certain but it might even be better to do it as an `eval` into LibGit2:
```julia
eval(LibGit2, quote
    function owner(x)
        depwarn("owner(x) is deprecated, use repository(x) instead.", :owner)
        repository(x)
    end
end)
```
I only mention the latter case because I'm not sure whether the fact that the symbol `:owner` isn't defined in the module in which `depwarn` is being called matters.